### PR TITLE
Feature/cleanup dockerfile

### PR DESCRIPTION
**Pull request by jbern0rd**
Mainly cleanup
Last question:
Do we remove the `Dockerfile` from `src/main/resources` or do we put it in this folder for all projects ?
It would be great to have a shared convention
What would you prefer ?

### CODE REVIEW

**Review by mcornaton — APPROVED** [review]
Thanks for this cleanup!
Regarding your question, I think I'd prefer the Dockerfile to be outside of the `src/main/resources` folder. It is not related to the SMS runtime so its place doesn't look to be there.

File: src/itest/java/com/iexec/sms/tee/GramineBeansLoadingTests.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 import org.springframework.test.context.ActiveProfiles;
 
 
-@ActiveProfiles(profiles = "gramine")
+@ActiveProfiles(profiles = {"gramine", "test"})
```
Comment (jeremyjams) [inline src/itest/java/com/iexec/sms/tee/GramineBeansLoadingTests.java:14]: Why not adding `"test"` to `TeeBeansLoadingTests`?

File: src/main/resources/Dockerfile [hunk 2]
```diff
@@ -5,12 +5,7 @@ ARG jar
 RUN test -n "$jar"
 
 COPY $jar                                       /app/iexec-sms.jar
-COPY src/main/resources/ssl-keystore-dev.p12    /app/ssl-keystore-dev.p12
 
-ENV IEXEC_SMS_SSL_KEYSTORE=/app/ssl-keystore-dev.p12
-ENV IEXEC_SMS_STORAGE_ENCRYPTION_AES_KEY_PATH=/scone/iexec-sms-aes.key
```
Comment (jeremyjams) [inline src/main/resources/Dockerfile:11]: Just a check, are values in core-ITs and elsewhere still good?

File: src/itest/java/com/iexec/sms/tee/GramineBeansLoadingTests.java (hunk 1 above)
Comment (jbern0rd) [reply src/itest/java/com/iexec/sms/tee/GramineBeansLoadingTests.java:14]: Good catch and I tried it. Sadly, this does not work.
Following the inheritance, profiles are appended and the last declared profile becomes kind of the main profile.
If "test" profile is applied on `TeeBeansLoadingTests`, the profiles list becomes ["mock-chain", "test", "gramine"]
In this case (and I don't know why), the `src/main/resource/application-gramine.yml` is being loaded and shadows the `src/itest/resources/application-test.yml` and tests are broken.
To stay efficient, I settled to the current implementation

Comment (jeremyjams) [reply src/itest/java/com/iexec/sms/tee/GramineBeansLoadingTests.java:14]: Ok :heavy_check_mark:

**Review by jeremyjams — APPROVED** [review]
Thanks :heavy_check_mark:

**Review by mcornaton — APPROVED** [review]

**Pull request merged**